.github/workflows/release-branch-tests.yml

```diff
@@ -0,0 +1,156 @@
+
+# **why?**
+# Ensures release branches and main are always shippable and not broken.
+# Also, can catch any dependencies shifting beneath us that might
+# introduce breaking changes (could also impact Cloud).
+
+# **when?**
+# Mainly on a schedule of 9:00, 13:00, 18:00 UTC everyday.
+# Manual trigger can also test on demand
+
+name: Release branch scheduled testing
+
+on:
+  workflow_call:
+    inputs:
+      workflows_to_run:
+        description: 'A list of workflow(s) to run aginst ex: ["main.yml, "integration.yml"]'
+        required: true
+        type: string
+
+# no special access is needed
+permissions: read-all
+
+jobs:
+  prep-work:
+    name: "Audit Inputs"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "[DEBUG] Print Inputs"
+        run: |
+          echo "workflows_to_run:    ${{ inputs.workflows_to_run }}"
+
+  fetch-latest-branches:
+    runs-on: ubuntu-latest
+
+    outputs:
+      branches_to_test: ${{ steps.add-main.outputs.full_branch_list }}
+
+    steps:
+      - name: "Fetch ${{ github.event.repository.name }} Latest Branches"
+        uses: dbt-labs/actions/fetch-repo-branches@v1.1.1
+        id: get-latest-branches
+        with:
+          repo_name: ${{ github.event.repository.name }}
+          organization: "dbt-labs"
+          pat: ${{ secrets.GITHUB_TOKEN }}
+          fetch_protected_branches_only: true
+          regex: "^1.[0-9]+.latest$"
+          perform_match_method: "match"
+          retries: 3
+
+      - name: "Add main to branch list"
+        id: add-main
+        run: |
+          full_branch_list="${{ steps.get-latest-branches.outputs.repo-branches }}"
+          # Swap single and double quotes because bash doesnt interpret correctly
+          full_branch_list="${full_branch_list//\'/__SINGLE_QUOTE__}"
+          full_branch_list="${full_branch_list//\"/\'}"
+          full_branch_list="${full_branch_list//__SINGLE_QUOTE__/\"}"
+          # now convert it to a bash array and add the main branch
+          full_branch_list=($(echo "$full_branch_list" | jq -r '.[]'))
+          for item in "${full_branch_list[@]}"; do   echo "$item"; done
+          full_branch_list+=("main")
+          # Convert the array to a string
+          full_branch_list_string=$(printf "'%s', " "${full_branch_list[@]}")
+          full_branch_list_string="[${full_branch_list_string%, }]"
+          # Assign the string to a variable
+          echo $full_branch_list_string
+          echo "full_branch_list=$full_branch_list_string" >> $GITHUB_OUTPUT
+
+      - name: "[ANNOTATION] ${{ github.event.repository.name }} - branches to test"
+        run: |
+          title="${{ github.event.repository.name }} - branches to test"
+          message="The workflow will run tests in ${{ inputs.workflows_to_run }} for the following branches of the ${{ github.event.repository.name }} repo: ${{ steps.add-main.outputs.full_branch_list }}"
+          echo "::notice $title::$message"
+
+  kick-off-ci:
```

Comment:
We could but I hesitate to do so.  This is a bigger conversation we need to have.  There are several schools of thought behind marketplace actions.  They can pose security risks since they've often so complex no one really know what the code does. They can also swap out the tags to do things you're not aware of.  They can be abandoned without notice.  Some shops just fork the action into their own repos and use that.  Some don't use them at all.  

The entire reason for this update is the action we were using was abandoned and we had no way to know.  This was true of multiple actions that we had to update as part of the deprecations work.

The actions managed by GitHub are generally considered safe (`actions/checkout`, `actions/setup-python`, etc) which makes sense.  Outside of that I think we need to come up with a standard.